packages/jflap-translator/src/convertJFLAP.ts

```diff
@@ -0,0 +1,160 @@
+
+// Define accepted project types
+import { xml2json } from 'xml-js'
+
+const PROJECT_TYPE_MAP = {
+  fa: 'FSA',
+  pda: 'PDA',
+  turing: 'TM'
+}
+
+export type GraphConfig = {
+  type: 'FSA' | 'tm' | 'pda'
+  statePrefix: string
+}
+
+/**
+ * Basic transition that stores common properties of other transition types
+ */
+export type Transition = {
+  id: number
+  from: number
+  to: number
+  read: string
+}
+
+export type PDATransition = Transition & {
+  push: string
+  pop: string
+}
+
+export type TMTransition = Transition & {
+  write: string
+  direction: string
+}
+
+/**
+ * Stores a state in a graph. Modeled after the frontend
+ */
+export type State = {
+  id: number
+  name: string
+  label: string
+  x: number
+  y: number
+  isFinal: boolean
+}
+
+/**
+ * Comment that is placed somewhere on the project
+ */
+export type Comment = {
+  id: number
+  x: number
+  y: number
+  text: string
+}
+
+// TODO: Replace with whatever type the frontenders come up with for the typescript conversion

```

Comment:
Note that this has been completed in the branch refactor/jstots with the Project type. Once feasible this type can be imported from types/ProjectTypes (new folder on refactor/jstots branch)

Reply:
Nice yeah meant to get back and update that. Will rebase and then use the types from the refactor.

How many stores are left? Would be good to either finish them off or just merge in what we have

Reply:
There's still 6 left but the remaining ones are significantly smaller than useProjectStore so will take less time. I think it could be worth merging in what we have so far though, because the project ones are pretty important and would be good to have them there without waiting for the other 6 (I should be able to get them done in the next few days I'm just a bit busy with other assignments atm). Should I open a PR to merge in what we have?

Reply:
Yeah would be good to merge in what we have

Then I can refactor the weird types in the simulator also to use what the frontend uses